### PR TITLE
Falsy File_Content

### PR DESCRIPTION
**Pull request by ehoanshelt**
I've been finding that I need a few checks to result in success if a regex pattern is found in a file. Adding falsy option to File_Content check allows me to do this.

- [ ]  @danielbachhuber

### CODE REVIEW

**Comment by ehoanshelt** [comment]
It looks like the check is failing on the WordPress checksum functionality.

https://travis-ci.org/wp-cli/doctor-command/jobs/218910257#L238-L244

Looks like something outside of what this PR is addressing

**Comment by danielbachhuber** [comment]
Can you include [functional tests](https://make.wordpress.org/cli/handbook/pull-requests/#running-and-writing-tests) for this change, please?

**Comment by danielbachhuber** [comment]
> I've been finding that I need a few checks to result in success if a regex pattern is found in a file.

I'm not 100% sure I follow. Do you mean you'd like for this check to assert the contents of a specific file, and error if the file contents don't include the value?

**Comment by ehoanshelt** [comment]
> Can you include functional tests for this change, please?

Would you like me to add these tests directly to features/check-file-contents.feature?

> I'm not 100% sure I follow. Do you mean you'd like for this check to assert the contents of a specific file, and error if the file contents don't include the value?

Correct. Did I miss something that could have done this without the falsy flag development?

**Comment by danielbachhuber** [comment]
> Would you like me to add these tests directly to features/check-file-contents.feature?

Sure!

> Correct. Did I miss something that could have done this without the falsy flag development?

I don't think so.

I'm not sure `$falsy` is the best argument name. What about calling it `$exists`, and defaulting to `$exists = false` ? Rationale being that you're checking whether something exists, and it should either exist or not exist.

**Comment by ehoanshelt** [comment]
I will get the tests done.

I like changing the name. I was trying to stay with the typical naming convention but $exists makes more sense. I will have those by early next week.

Thanks for the help so far!

**Comment by danielbachhuber** [comment]
> I will get the tests done.

Sounds good.

> Thanks for the help so far!

Yep!

**Comment by schlessera** [comment]
@ehoanshelt We've done changes to the testing code. Can you please merge the latest master into this PR?

**Comment by ehoanshelt** [comment]
@schlessera Latest master has been merged into this PR. I believe I have addressed the rest of the concerns. Please let me know if I need to update anything else!

File: features/check-file-contents.feature [hunk 1]
```diff
@@ -107,3 +107,33 @@ Feature: Check files in a WordPress install
...
+        options:
+          regex: .*wp-doctor-exists-test.*
+          only_wp_content: true
+          exists: true
+      """
+
+    When I run `wp doctor check file-content-exist --config=config.yml --format=json`
+    Then STDOUT should be JSON containing:
+      """
+      [{"name":"file-content-exist","status":"error","message":"0 'php' files failed check for '.*wp-doctor-exists-test.*'."}]
```
Comment (danielbachhuber) [inline features/check-file-contents.feature]: Is this count correct?

```
0 'php' files failed check
```

File: inc/checks/class-file-contents.php [hunk 2]
```diff
@@ -15,18 +15,40 @@ class File_Contents extends File {
 	 * @var string
 	 */
 	protected $regex;
+	protected $exists = false;
```
Comment (danielbachhuber) [inline inc/checks/class-file-contents.php]: Can you include PHPdoc for this class variable?

File: inc/checks/class-file-contents.php [hunk 3]
```diff
@@ -15,18 +15,40 @@ class File_Contents extends File {
...
 	public function run() {
 
 		if ( isset( $this->regex ) ) {
 			if ( ! empty( $this->_matches ) ) {
-				$this->set_status( 'error' );
-				$count = count( $this->_matches );
-				$message = $count === 1 ? "1 '{$this->extension}' file" : "{$count} '{$this->extension}' files";
-				$this->set_message( "{$message} failed check for '{$this->regex}'." );
+				//if matches are found
+				if ( $this->exists == true ) {
```
Comment (danielbachhuber) [inline inc/checks/class-file-contents.php]: This needs to be a yoda condition

File: inc/checks/class-file-contents.php [hunk 4]
```diff
@@ -15,18 +15,40 @@ class File_Contents extends File {
...
+					//$exists set to true so we should report true if something is found
+					$this->set_status( 'success' );
+					$count = count( $this->_matches );
+					$message = $count === 1 ? "1 '{$this->extension}' file" : "{$count} '{$this->extension}' files";
```
Comment (danielbachhuber) [inline inc/checks/class-file-contents.php]: Needs to be yoda condition

File: inc/checks/class-file-contents.php [hunk 5]
```diff
@@ -15,18 +15,40 @@ class File_Contents extends File {
...
+					$this->set_message( "{$message} passed check for '{$this->regex}'." );
+				}else{
+					//$exists is not set to true so we should report error if something is found
+					$this->set_status( 'error' );
+					$count = count( $this->_matches );
+					$message = $count === 1 ? "1 '{$this->extension}' file" : "{$count} '{$this->extension}' files";
```
Comment (danielbachhuber) [inline inc/checks/class-file-contents.php]: Needs to be yoda condition

File: inc/checks/class-file-contents.php [hunk 6]
```diff
@@ -15,18 +15,40 @@ class File_Contents extends File {
...
-				$message = $count === 1 ? "1 '{$this->extension}' file" : "{$count} '{$this->extension}' files";
-				$this->set_message( "{$message} failed check for '{$this->regex}'." );
+				//if matches are found
+				if ( $this->exists == true ) {
+					//$exists set to true so we should report true if something is found
+					$this->set_status( 'success' );
+					$count = count( $this->_matches );
+					$message = $count === 1 ? "1 '{$this->extension}' file" : "{$count} '{$this->extension}' files";
+					$this->set_message( "{$message} passed check for '{$this->regex}'." );
+				}else{
```
Comment (danielbachhuber) [inline inc/checks/class-file-contents.php]: Spacing

File: inc/checks/class-file-contents.php [hunk 7]
```diff
@@ -15,18 +15,40 @@ class File_Contents extends File {
...
+					$this->set_status( 'error' );
+					$count = count( $this->_matches );
+					$message = $count === 1 ? "1 '{$this->extension}' file" : "{$count} '{$this->extension}' files";
+					$this->set_message( "{$message} failed check for '{$this->regex}'." );
+				}
 			} else {
-				$this->set_status( 'success' );
-				$this->set_message( "All '{$this->extension}' files passed check for '{$this->regex}'." );
+				//No Matches Found
+				if ( $this->exists == true ) {
```
Comment (danielbachhuber) [inline inc/checks/class-file-contents.php]: Needs to be yoda condition

File: inc/checks/class-file-contents.php [hunk 8]
```diff
@@ -15,18 +15,40 @@ class File_Contents extends File {
...
+					//$exists set to true so we should report error if regex is not found
+					$this->set_status( 'error' );
+					$count = count( $this->_matches );
+					$message = $count === 1 ? "1 '{$this->extension}' file" : "{$count} '{$this->extension}' files";
```
Comment (danielbachhuber) [inline inc/checks/class-file-contents.php]: Needs to be yoda condition

File: inc/checks/class-file-contents.php [hunk 9]
```diff
@@ -15,18 +15,40 @@ class File_Contents extends File {
...
+					$this->set_message( "{$message} passed check for '{$this->regex}'." );
+				}else{
```
Comment (danielbachhuber) [inline inc/checks/class-file-contents.php]: Spacing

File: features/check-file-contents.feature (hunk 1 above)
Comment (ehoanshelt) [inline features/check-file-contents.feature]: The count appears to be correct but the messaging is not, which I believe is causing the error. Looking into it now

File: inc/checks/class-file-contents.php (hunk 2 above)
Comment (ehoanshelt) [inline inc/checks/class-file-contents.php]: Done

File: inc/checks/class-file-contents.php (hunk 3 above)
Comment (ehoanshelt) [inline inc/checks/class-file-contents.php]: Done

File: inc/checks/class-file-contents.php (hunk 4 above)
Comment (ehoanshelt) [inline inc/checks/class-file-contents.php]: Done

File: inc/checks/class-file-contents.php (hunk 6 above)
Comment (ehoanshelt) [inline inc/checks/class-file-contents.php]: Done

File: inc/checks/class-file-contents.php (hunk 5 above)
Comment (ehoanshelt) [inline inc/checks/class-file-contents.php]: conditional not needed. Removed.

File: inc/checks/class-file-contents.php (hunk 7 above)
Comment (ehoanshelt) [inline inc/checks/class-file-contents.php]: Done

File: inc/checks/class-file-contents.php (hunk 8 above)
Comment (ehoanshelt) [inline inc/checks/class-file-contents.php]: Done

File: inc/checks/class-file-contents.php (hunk 9 above)
Comment (ehoanshelt) [inline inc/checks/class-file-contents.php]: Done

**Comment by ehoanshelt** [comment]
@danielbachhuber I believe I have hit all your points from the last review. :-) Let me know if I missed or need anything else.

**Comment by danielbachhuber** [comment]
> Let me know if I missed or need anything else.

Looks good. Further clarified the PHPdoc comment in a9432b0d78e836cda045083d27b5e01a509a970e

**Pull request merged**